syntax = "proto3";

package zeva.transactions;
import "common.proto";

service ZEVModels {
  rpc ListModels (ModelListRequest) returns (stream ModelSummary);
  rpc GetModelDetails(ModelDetailsRequest) returns (ModelDetails);
  rpc CreateModelRequest(ModelCreateRequest) returns (ModelCreateResponse);
  rpc UpdateModelRequest(ModelUpdateRequest) returns (ModelUpdateResponse);
}

message ModelListRequest {
    uint64 organization = 1;
}

message ModelSummary {
    // what we want to see in a list view
    uint64 id = 1;
    string make = 2;
    string model = 3;
    string type = 4;
    string trim = 5;
    uint32 range = 6;
    CreditValue credits = 7;
}

message ModelDetails {
    // what we want to see in a detail view
    uint64 id = 1;
    string make = 2;
    string model = 3;
    string type = 4;
    string trim = 5;
    uint32 range = 6;
    CreditValue credits = 7;
}

message ModelDetailsRequest {
    uint64 id = 1;
}

message ModelCreateRequest {
    string make = 2;
    string model = 3;
    string type = 4;
    string trim = 5;
    uint32 range = 6;
}

message ModelCreateResponse {
    uint64 id = 1;
}

message ModelUpdateRequest {
    uint64 id = 1;
    string make = 2;
    string model = 3;
    string type = 4;
    string trim = 5;
    uint32 range = 6;
}

message ModelUpdateResponse {
}
